### DISCUSSION TITLE
A warning in browser console

### DISCUSSION THREAD

**Discussion by alecpl** [openpgpjs/openpgpjs#1414] [Q&A]
When loading openpgp.min.js v5.0.0 in Firefox browser I see three times this warning: "asm.js type error: Disabled because no suitable wasm compiler is available". What are the implications and should the warning exist at all?

**Comment by twiss** [comment]
Hello :wave: This is more of a question for Firefox, but AFAIU this only happens when you have the Developer Tools open. Indeed if I load OpenPGP.js first and _then_ open the devtools, I see "Successfully compiled asm.js code" * 3 instead. Therefore, this shouldn't be a big deal.
Also, even if wasm compilation fails, it will still work (due to asm.js being backwards compatible with js), but there might be some increased concern about timing attacks. (Even though wasm also makes no guarantees about timing, at least until constant-time wasm progresses - but they may be more difficult to exploit in wasm.)
